src/eei.cpp

```diff
@@ -239,6 +254,9 @@ namespace HeraVM {
 
       HERA_DEBUG << "codeCopy " << hex << resultOffset << " " << codeOffset << " " << length << dec << "\n";
 
+      heraAssert(ffs(GasSchedule::copy) + (ffs(length) - 5) <= 64, "Gas charge overflow");
+      heraAssert(numeric_limits<uint64_t>::max() - GasSchedule::stepGas2 >= GasSchedule::copy * (length / 32), "Gas charge overflow");
+      takeGas((uint64_t)ceil(GasSchedule::stepGas2 + GasSchedule::copy * (length / 32)));
```

Comment:
This one uses `ceil`, but the one in `externalCodeCopy` doesn't. Any reason for that? Also shouldn't this instead just use `(length + 31) / 32` to avoid the use of `ceil`?

Reply:
It also may make sense creating a helper along the lines of `takeWordCopyGas(baseCost, costPerWord, length)`

Reply:
I will do this in a separate PR if there is a significant need for it.